### PR TITLE
114 dicom metadata

### PR DESCRIPTION
**Pull request by lillie-dae**
Changes required for #114 
Also includes example integration test.

### Description

Providing DICOM metadata from Workflow Usage.

### Status
**In Progress**

### Types of changes
- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [x] All tests passed locally.
- [ ] [Documentation comments](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/documentation-comments) included/updated.
- [ ] User guide updated.
- [ ] I have updated the changelog
- [x] Any new files have copyright headers
- [ ] Code coverage above 70%
- [x] TODO items resolved

### CODE REVIEW

File: src/Monai.Deploy.WorkflowManager.Storage/Services/DicomService.cs [hunk 1]
```diff
@@ -106,5 +106,140 @@ public async Task<IEnumerable<string>> GetDicomPathsForTask(string outputDirecto
...
+        //    }
+
+        //    foreach (var item in items)
+        //    {
+        //        if (!item.FilePath.EndsWith(".dcm.json"))
+        //        {
+        //            continue;
+        //        }
+
+        //        var stream = await _storageService.GetObjectAsync(bucketId, $"{payloadId}/dcm/{item.Filename}");
```
Comment (lillie-dae) [inline src/Monai.Deploy.WorkflowManager.Storage/Services/DicomService.cs:221]: going to remove commented code

File: src/WorkflowExecuter/Services/IWorkflowExecuterService.cs [hunk 2]
```diff
@@ -19,7 +19,6 @@ public interface IWorkflowExecuterService
         /// <param name="message">The task update message event.</param>
         Task<bool> ProcessTaskUpdate(TaskUpdateEvent message);
 
-        /// <summary>
```
Comment (lillie-dae) [inline src/WorkflowExecuter/Services/IWorkflowExecuterService.cs:22]: missing         /// <summary>

File: src/WorkflowExecuter/Services/WorkflowExecuterService.cs [hunk 3]
```diff
@@ -178,20 +181,23 @@ public async Task<bool> ProcessTaskUpdate(TaskUpdateEvent message)
...
             {
-                var dicomImages = (await _dicomService.GetDicomPathsForTask(currentTask.OutputDirectory, workflowInstance.BucketId))?.ToList();
+                var dicomImages = (await _dicomService.GetDicomPathsForTaskAsync(currentTask.OutputDirectory, workflowInstance.BucketId))?.ToList();
 
                 if (dicomImages is not null && dicomImages.Any())
                 {
                     return await HandleDicomExport(workflowInstance, currentTask, exportDestinations, dicomImages, message.CorrelationId);
                 }
-
-                _logger.ExportFilesNotFound(currentTask.TaskId, workflowInstance.Id);
```
Comment (lillie-dae) [inline src/WorkflowExecuter/Services/WorkflowExecuterService.cs:194]: i think this was a rebase/merge conflict mistake ill add it back in

File: src/WorkflowExecuter/Services/WorkflowExecuterService.cs [hunk 4]
```diff
@@ -348,7 +354,7 @@ private async Task<bool> HandleTaskDestinations(WorkflowInstance workflowInstanc
             return processed;
         }
 
-        private async Task<List<TaskExecution>> CreateTaskDestinations(WorkflowInstance workflowInstance, WorkflowRevision workflow, string taskId)
+        public async Task<List<TaskExecution>> CreateTaskDestinations(WorkflowInstance workflowInstance, WorkflowRevision workflow, string taskId)
```
Comment (lillie-dae) [inline src/WorkflowExecuter/Services/WorkflowExecuterService.cs:357]: doesn't need to be public changing it back to private

File: tests/UnitTests/WorkflowExecuter.Tests/Services/ConditionalParameterParserTests.cs [hunk 5]
```diff
@@ -0,0 +1,451 @@
...
+
+
+
+            Assert.Equal(expectedResult, actualResult);
+        }
+
+        public List<WorkflowInstance> CreateTestData()
+        {
+            //  {{context.executions.task['TaskID'].'Key'}}
+            //  {{context.executions.task['2dbd1af7-b699-4467-8e99-05a0c22422b4'].'Fred'}}
```
Comment (jackschofield23) [inline tests/UnitTests/WorkflowExecuter.Tests/Services/ConditionalParameterParserTests.cs:68]: comments left in

File: src/WorkflowExecuter/Services/WorkflowExecuterService.cs [hunk 6]
```diff
@@ -63,6 +65,7 @@ public class WorkflowExecuterService : IWorkflowExecuterService
             _workflowRepository = workflowRepository ?? throw new ArgumentNullException(nameof(workflowRepository));
             _workflowInstanceRepository = workflowInstanceRepository ?? throw new ArgumentNullException(nameof(workflowInstanceRepository));
             _messageBrokerPublisherService = messageBrokerPublisherService ?? throw new ArgumentNullException(nameof(messageBrokerPublisherService));
+            _conditionalParameterParser = conditionalParser;
```
Comment (jackschofield23) [inline src/WorkflowExecuter/Services/WorkflowExecuterService.cs:68]: Add null check

**Review by jackschofield23 — APPROVED** [review]

**Pull request merged**